lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -294,8 +308,29 @@ def write(chunk)
             end
           end
         end
+        if @running_cloudfunctions
+          # If the current group of entries is coming from a Cloud Functions
+          # function, the function name can be extracted from the tag.
+          match_data = @cloudfunctions_tag_regexp.match(tag)
+          if match_data
+            # Service name is set to Cloud Functions only for logs actually
+            # coming from a function.
+            @service_name = CLOUDFUNCTIONS_SERVICE
+            labels = write_log_entries_request['commonLabels']
+            labels["#{CLOUDFUNCTIONS_SERVICE}/function_name"] =
+              match_data['function_name']
+          else
+            # Other logs are considered as coming from the Container Engine
+            # service.
+            @service_name = CONTAINER_SERVICE
```

Comment:
This is run in a loop (line 294) so service_name could be set to CLOUDFUNCTIONS_SERVICE in previous iteration while we want it to be CONTAINER_SERVICE in this iteration. The general problem here is that I want the service_name to depend on the value of tag so it can change with every loop iteration. I could potentially introduce a local variable like actual_service_name, set it inside the loop and use it when creating a log entry - but I am not sure if this wouldn't be more confusing after all.